diem-node/src/lib.rs

```diff
@@ -341,6 +354,7 @@ pub fn setup_environment(node_config: &NodeConfig, logger: Option<Arc<Logger>>)
             node_config.base.role,
             network_config,
             TimeService::real(),
+            Some(&mut event_subscription_service),
```

Comment:
seems weird some places we pass the service, some places we pass the subscription

Reply:
@zekun000 -- see my note in the description:
> I'm not proud of how I managed to hook up the network builder (i.e., in commit 3), but it seems to be okay. The network folks might want to re-think this in the future.

The reason for this is because we don't know many subscribers the network builder is going to require. It's not great, and we can probably clean it up in the future, but it is what it is for now 😄

Reply:
The network builder needs an overhaul in general because it's been built in a weird way.  I originally thought it was better than the original version, but it needs some more love.